### PR TITLE
Pylint cleanup and metdata/attachment methods

### PR DESCRIPTION
**Pull request by xmunoz**
Needs tests...

### CODE REVIEW

**Comment by xmunoz** [comment]
@whroman, can you take a look?

File: sodapy/__init__.py [hunk 1]
```diff
@@ -305,8 +338,35 @@ def _raise_for_status(response):
...
     for param in args:
         if args[param] is not None:
             result[param] = args[param]
     return result
+
+def authentication_validation(username, password, access_token):
+    '''
+    Only accept one form of authentication.
+    '''
+    if bool(username) != bool(password):
```
Comment (whroman) [inline sodapy/__init__.py]: `!=` is used here, but `is not` is used elsewhere, like L346. Pick one ;)

Comment (xmunoz) [inline sodapy/__init__.py]: This is a quirk of python. I guess in this case I am actually comparing singleton values, so `is not` would be more appropriate than `!=` according to PEP8. I guess the best thing would probably be to simulate a nand by employing some kind of `not ( x and y)`. Will update.

**Comment by timwis** [comment]
This looks awesome @xmunoz! Thanks for taking the time to make these improvements

**Pull request merged**